polynote-kernel/src/main/scala/polynote/data/Rope.scala

```diff
@@ -114,7 +114,7 @@ object Rope {
 
   /** Creates a rope from a character array. */
   def apply(a: Array[Char]): Rope =
-    if (a == null || a.isEmpty) {
+    if (a == null || a.length == 0) {
```

Comment:
Could you explain the reasoning behind this change?

Reply:
Polynote build fails in my environment on this point. As far as i understood isEmpty needs an implicit conversion, and there is a conflict in the implicit resolution with JDK15+, see also https://github.com/scala/bug/issues/12172 and the proposed simple workaround https://github.com/scala/bug/issues/12172#issuecomment-706712575.

Reply:
Gotcha - we develop internally with Polynote on JDK 8, which is probably the reason for this issue. Is it possible to try this out local on JDK 8? We'd prefer to not have to change this if possible, but I can also submit the change to your branch if not.

Reply:
Yes, this works on JDK 8, but JDK 8 is outdated in most of the customer environments i work. In my opinion a Polynote build should also work with current JDKs, but you can compile with compatibility settings for JDK 8, e.g. targetCompatibility 1.8. What do you think?